src/eth/GasEstimatorService.js

```diff
@@ -1,54 +1,114 @@
-import { PrivateService } from '@makerdao/services-core';
+import { PublicService } from '@makerdao/services-core';
+import fetch from 'isomorphic-fetch';
 
-export default class GasEstimatorService extends PrivateService {
+export default class GasEstimatorService extends PublicService {
   constructor(name = 'gasEstimator') {
     super(name, ['web3', 'log']);
+    this._fallback = 4000000;
     this._multiplier = 1.55;
-    this._absolute = null;
+    this._transactionSpeed = 'fast';
+  }
+
+  initialize(settings) {
+    this._gasStationData = this.fetchGasStationData();
+
+    if (settings) {
+      this._parseConfig(settings.gasLimit, 'gasLimit');
+      this._parseConfig(settings.gasPrice, 'gasPrice');
+    }
+  }
+
+  _parseConfig(settings = 'default', label) {
+    return settings === 'default' || typeof settings === 'object'
+      ? this._setProperties(settings, label)
+      : (this[label] = settings);
   }
 
-  authenticate() {
-    const settings = this.get('web3').transactionSettings();
-    this._fallback = settings && settings.gasLimit
-      ? settings.gasLimit
-      : 4000000;
+  _setProperties(settings, label) {
+    if (settings === 'default') return;
+
+    return Object.keys(settings).map(key => {
```

Comment:
since we have lodash in this project you could also use its [map](https://devdocs.io/lodash~4/index#map) function:
```js
import { map } from 'lodash';
map(settings, (value, key) => { ... })
```

Reply:
Nice, good call 👍